saw-core/src/Verifier/SAW/Term/Pretty.hs

```diff
@@ -304,18 +321,20 @@ ppAppList _ f [] = f
 ppAppList p f args = ppParensPrec p PrecApp $ group $ hang 2 $ vsep (f : args)
 
 -- | Pretty-print "let x1 = t1 ... xn = tn in body"
-ppLetBlock :: [(MemoVar,Doc)] -> Doc -> Doc
+ppLetBlock :: [(MemoVar, Doc)] -> Doc -> Doc
 ppLetBlock defs body =
-  text "let" <+> lbrace <+> align (vcat (map ppEqn defs)) <$$>
-  indent 4 rbrace <$$>
-  text " in" <+> body
+  vcat
+  [ text "let" <+> lbrace <+> align (vcat (map ppEqn defs))
```

Comment:
If you enable `OverloadedStrings`, then you don't even need the `text` (aka `pretty`) here:  `[ "let" <+> lbrace ...`

It's up to you which form you prefer, just pointing out the possibility.  And of course, sometimes `OverloadedStrings` has other consequences.

Reply:
I actually considered doing the overloaded strings trick, but in the end I chose to define `text` and friends to minimize the diff, essentially. There are a bunch of other places where `text` was used with strings that were not string literals, so it felt more consistent to me to just keep them all. If I was writing all the pretty printing code from scratch using `prettyprinter`, it might have looked different.

Reply:
As of 056ac76c3ba0cdafe0ea054131da8d48660448c9, I got rid of the locally-defined type-specific combinators. The code now looks more idiomatic for `prettyprinter`.